## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 479302`
Commit: [`1cabb030eafad0daef42bbde8affaf2a4b99e8e7`](https://github.com/NixOS/nixpkgs/commit/1cabb030eafad0daef42bbde8affaf2a4b99e8e7) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/1cabb030eafad0daef42bbde8affaf2a4b99e8e7..pull/479302/head))
Merge: [`2b22151a2e90a8ff35508c637d63b2be6a608090`](https://github.com/NixOS/nixpkgs/commit/2b22151a2e90a8ff35508c637d63b2be6a608090)

Logs: https://github.com/qbisi/nixpkgs-review-gha/actions/runs/20916589166


---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>miktex</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>miktex</summary>
<pre>[ 31%] Generating hibasetypes.h, hierror.h, hiformat.c, hiformat.h, higet.c, higet.h, hilexer.l, himktables.c, hiparser.y, hiput.c, hiput.h, hishrink.c, hiskip.c, histretch.c
miktex-ctangle: major issue: So far, you have not checked for MiKTeX updates.
[ 31%] Building CXX object Programs/TeXAndFriends/hitex/CMakeFiles/himktables.dir/__/__/__/Libraries/MiKTeX/etc/wrapper.cpp.o
[ 31%] Building C object Programs/TeXAndFriends/hitex/CMakeFiles/himktables.dir/himktables.c.o
[ 31%] Linking CXX executable ../../../sandbox/miktex/bin/linux-x86_64/himktables
[ 31%] Built target himktables
[ 31%] Generating hitex.c
miktex-ctangle: major issue: So far, you have not checked for MiKTeX updates.
[ 31%] Generating hitables.c
himktables: major issue: So far, you have not checked for MiKTeX updates.
[ 31%] Building C object Programs/TeXAndFriends/hitex/CMakeFiles/miktex-hitex.dir/hiput.c.o
[ 31%] Building C object Programs/TeXAndFriends/hitex/CMakeFiles/miktex-hitex.dir/hitables.c.o
[ 31%] Building C object Programs/TeXAndFriends/hitex/CMakeFiles/miktex-hitex.dir/hitex.c.o
/build/source/build/Programs/TeXAndFriends/hitex/hitex.c:88:10: fatal error: hb.h: No such file or directory
   88 | #include &quot;hb.h&quot;
      |          ^~~~~~
compilation terminated.
make[2]: *** [Programs/TeXAndFriends/hitex/CMakeFiles/miktex-hitex.dir/build.make:160: Programs/TeXAndFriends/hitex/CMakeFiles/miktex-hitex.dir/hitex.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:7217: Programs/TeXAndFriends/hitex/CMakeFiles/miktex-hitex.dir/all] Error 2
make: *** [Makefile:166: all] Error 2</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>miktex</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>miktex</summary>
<pre>[ 31%] Generating hibasetypes.h, hierror.h, hiformat.c, hiformat.h, higet.c, higet.h, hilexer.l, himktables.c, hiparser.y, hiput.c, hiput.h, hishrink.c, hiskip.c, histretch.c
miktex-ctangle: major issue: So far, you have not checked for MiKTeX updates.
[ 31%] Building CXX object Programs/TeXAndFriends/hitex/CMakeFiles/himktables.dir/__/__/__/Libraries/MiKTeX/etc/wrapper.cpp.o
[ 31%] Building C object Programs/TeXAndFriends/hitex/CMakeFiles/himktables.dir/himktables.c.o
[ 31%] Linking CXX executable ../../../sandbox/miktex/bin/linux-aarch64/himktables
[ 31%] Built target himktables
[ 31%] Generating hitex.c
miktex-ctangle: major issue: So far, you have not checked for MiKTeX updates.
[ 31%] Generating hitables.c
himktables: major issue: So far, you have not checked for MiKTeX updates.
[ 31%] Building C object Programs/TeXAndFriends/hitex/CMakeFiles/miktex-hitex.dir/hiput.c.o
[ 31%] Building C object Programs/TeXAndFriends/hitex/CMakeFiles/miktex-hitex.dir/hitables.c.o
[ 31%] Building C object Programs/TeXAndFriends/hitex/CMakeFiles/miktex-hitex.dir/hitex.c.o
/build/source/build/Programs/TeXAndFriends/hitex/hitex.c:88:10: fatal error: hb.h: No such file or directory
   88 | #include &quot;hb.h&quot;
      |          ^~~~~~
compilation terminated.
make[2]: *** [Programs/TeXAndFriends/hitex/CMakeFiles/miktex-hitex.dir/build.make:160: Programs/TeXAndFriends/hitex/CMakeFiles/miktex-hitex.dir/hitex.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:7217: Programs/TeXAndFriends/hitex/CMakeFiles/miktex-hitex.dir/all] Error 2
make: *** [Makefile:166: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/mi/miktex/package.nix
+++ b/pkgs/by-name/mi/miktex/package.nix
@@ -29,7 +29,6 @@
   gd,
   gmp,
   graphite2,
-  harfbuzzFull,
   hunspell,
   libjpeg,
   log4cxx,
@@ -94,13 +93,13 @@ let
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "miktex";
-  version = "25.4";
+  version = "25.12";
 
   src = fetchFromGitHub {
     owner = "miktex";
     repo = "miktex";
     tag = finalAttrs.version;
-    hash = "sha256-3QGW8rsettA+Jtrsi9C5ONIG4vP+iuUEUi9dGHfWMSY=";
+    hash = "sha256-gg6qTyY1LkHcUxTQN92yYark0uVoQjXDTbTMD6BucXA=";
   };
 
   patches = [
@@ -136,12 +135,6 @@ stdenv.mkDerivation (finalAttrs: {
 
     substituteInPlace Programs/TeXAndFriends/omega/otps/source/outocp.c \
       --replace-fail 'fprintf(stderr, s);' 'fprintf(stderr, "%s", s);'
-  ''
-  # This patch fixes mismatch char types (signed int and unsigned int) on aarch64-linux platform.
-  # Should not be applied to other platforms otherwise the build will fail.
-  + lib.optionalString (stdenv.hostPlatform.isAarch64 && stdenv.hostPlatform.isLinux) ''
-    sed -i 's/--using-namespace=MiKTeX::TeXAndFriends/& --chars-are-unsigned/g' \
-      Programs/TeXAndFriends/Knuth/web/CMakeLists.txt
   '';
 
   strictDeps = true;
@@ -173,7 +166,6 @@ stdenv.mkDerivation (finalAttrs: {
     gd
     gmp
     graphite2
-    harfbuzzFull
     hunspell
     libjpeg
     log4cxx
@@ -194,6 +186,8 @@ stdenv.mkDerivation (finalAttrs: {
     (lib.cmakeBool "WITH_BOOTSTRAPPING" true)
     (lib.cmakeBool "USE_SYSTEM_POPPLER" true)
     (lib.cmakeBool "USE_SYSTEM_POPPLER_QT" true)
+    (lib.cmakeBool "USE_SYSTEM_HARFBUZZ" false)
+    (lib.cmakeBool "USE_SYSTEM_HARFBUZZ_ICU" false)
     (lib.cmakeBool "MIKTEX_SELF_CONTAINED" false)
     # Miktex infers install prefix by stripping CMAKE_INSTALL_BINDIR from the called program.
     # It should not be set to absolute path in default cmakeFlags, otherwise an infinite loop will happen.
@@ -206,6 +200,9 @@ stdenv.mkDerivation (finalAttrs: {
   env = {
     LANG = "C.UTF-8";
     MIKTEX_REPOSITORY = "file://${miktexLocalRepository}/";
+    # force char to be unsigned on none x86 systems
+    # See https://github.com/MiKTeX/miktex/issues/1440
+    NIX_CFLAGS_COMPILE = "-fsigned-char";
   };
 
   enableParallelBuilding = false;

